Docs (#42)
SoftwareForScience/jiskefet: docs/ARCHITECTURE.md

## Application

### Use Cases
The software in this layer contains application specific business rules. It encapsulates and implements all of the use cases of the system. These use cases orchestrate the flow of data to and from the entities, and direct those entities to use their enterprise wide business rules to achieve the goals of the use case.

We do not expect changes in this layer to affect the entities. We also do not expect this layer to be affected by changes to externalities such as the database, the UI, or any of the common frameworks. This layer is isolated from such concerns.

We do, however, expect that changes to the operation of the application will affect the use-cases and therefore the software in this layer. If the details of a use-case change, then some code in this layer will certainly be affected.

### Framework

> This layer is where all the I/O components go: the UI, database, frameworks, devices, etc. It's the most volatile layer. Since the things in this layer are so likely to change, they are kept as far away as possible from the more stable domain layers. Because they are kept separate, it's relatively easy make changes or swap one component for another.

Reviewer: ```suggestion
This layer is where all the I/O components go: the UI, database, frameworks, devices, etc. It's the most volatile layer. Since these components are so likely to change, they are kept as far away, from the more stable domain layers, as possible. Because they are kept separate, it's relatively easy make changes or swap one component for another.
```
